json-schema/json-schema-circe/src/test/scala/endpoints/circe/JsonSchemasTest.scala

```diff
@@ -59,6 +59,38 @@ class JsonSchemasTest extends AnyFreeSpec {
     )
   }
 
+  "sealed trait tagged merge" in {
+    val barJson =
+      Json.obj(
+        "type" -> Json.fromString("Bar"),
+        "s" -> Json.fromString("foo")
+      )
+    val bar = Bar("foo")
+    assert(Foo.alternativeSchemaForMerge.decoder.decodeJson(barJson).exists(_ == bar))
+    assert(Foo.alternativeSchemaForMerge.encoder.apply(bar) == barJson)
+
+    assert(
+      Foo.alternativeSchemaForMerge.decoder
+        .decodeJson(Json.obj())
+        .swap
+        .exists(
+          _ == DecodingFailure("Missing type discriminator field 'type'!", Nil)
+        )
+    )
+    val wrongJson = Json.obj(
+      "type" -> Json.fromString("Unknown"),
+      "s" -> Json.fromString("foo")
+    )
+    assert(
+      Foo.alternativeSchemaForMerge.decoder
+        .decodeJson(wrongJson)
+        .swap
+        .exists(
+          _ == DecodingFailure("No decoder for discriminator 'Unknown'!", Nil)
+        )
+    )
+  }
+
```

Comment:
Thanks for writing these tests! If possible, we should first try to write tests in the [endpoints.algebra.JsonSchemasOptionalFieldsTest](https://github.com/julienrf/endpoints/blob/master/json-schema/json-schema/src/test/scala/endpoints/algebra/JsonSchemasOptionalFieldsTest.scala) file so that the tests are written once and reused by all the interpreters. (The file is incorrectly named, it contains tests for other things than just optional fields…) Can you check if it is possible to move these tests there?

Reply:
Yes I'll check if I can move them

Reply:
Mm, it seems the JsonSchemasOptionalFieldsTest contains different kind of tests. There isn't a single tagged test there. I think the test above is quite inline with the other tagged test above. Also the schema created using the `orElseMerge` is in the same fixtures file as the alternative using `orElse`. I might be missing something but it seems to me the tests are added in a similar fashion as the other tagged tests that already existed.

Reply:
OK, no problem to keep the tests as you implemented them, then!